conans/util/windows.py

```diff
@@ -58,8 +61,24 @@ def path_shortener(path, short_paths):
 
     short_home = os.getenv("CONAN_USER_HOME_SHORT")
     if not short_home:
-        drive = os.path.splitdrive(path)[0]
-        short_home = os.path.join(drive, os.sep, ".conan")
+        if OSInfo().is_cygwin:
+            try:
+                cmd = ['cygpath', path, '--unix']
```

Comment:
I would really prefer to run this code just once, to define the "short home" folder, than to launch this subprocess for every short path. All the shortened paths should be shortened to the same short home, so it seems that evaluating this once is enough.

Reply:
That would be a feature/fix not only for Cygwin but for Windows too. If we forget about environment variables, this function creates the short_path in the same _drive_ as the input `path` parameter, and it can be different for every `path`.

I agree with you and we should probably compute the `short_home` value only once based on the cache folder/drive. Probably it is something to retrieve/store in the `config` object like `config.short_home` and forget about all the logic in this function. Probably better for another PR, don't you think so?

Reply:
For the record, this makes sense to me.

Reply:
Issue here: https://github.com/conan-io/conan/issues/6765